qiskit/pulse/instructions/instruction.py

```diff
@@ -131,7 +137,7 @@ def duration(self) -> int:
         return self._duration
 
     @property
-    def _children(self) -> Tuple[ScheduleComponent]:
+    def children(self) -> Tuple[ScheduleComponent]:
```

Comment:
This should remain private

Reply:
It needs to be manipulated by transforms in a separate module, therefore it cannot remain private.

Reply:
It should not be manipulated by a separate module. If it seems like it's needed, then `Schedule` needs new methods. We need to be able to modify the way data is structured internally without changing the user interface.

Reply:
I agree. I think `instructions` and `_children` should be equivalent. Embedding a `Schedule` in a `Schedule` should be done with a `Call(schedule)` instruction. To obtain the current behavior of instructions there should exist transform (or a compiler pass) `flatten` which returns a new schedule with all `Call ` instructions inlined and unrolled.

If you agree with the (not saying you should 😄 ) the question is how should we proceed right now. Have transforms manipulate `_children`? Or make `_children` public as `children`?